src/Adapter/Customer/Group/QueryHandler/GetCustomerGroupForEditingHandler.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Adapter\Customer\Group\QueryHandler;
+
+
+use Group as CustomerGroup;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Exception\GroupNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Query\GetCustomerGroupForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\QueryHandler\GetCustomerGroupForEditingHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\QueryResult\EditableCustomerGroup;
+
+class GetCustomerGroupForEditingHandler implements GetCustomerGroupForEditingHandlerInterface
+{
+    /**
+     * @param GetCustomerGroupForEditing $query
+     *
+     * @throws GroupNotFoundException
+     *
+     * @return EditableCustomerGroup
+     */
+    public function handle(GetCustomerGroupForEditing $query): EditableCustomerGroup
+    {
+        $customerGroupId = $query->getCustomerGroupId();
+        $customerGroup = new CustomerGroup($customerGroupId->getValue());
+
+        if ($customerGroup->id !== $customerGroupId->getValue()) {
```

Comment:
there is GroupRepository, so you should inject it and use $this->groupRepository->get($customerGroupId). 
which internally would check if group was found.

Reply:
Indeed, you shouldn't manipulate legacy ObjectModel directly like this, prefer using the ObjectModelRepository instead (especially since you already created it), even if it's not perfect yet it's a first step to using future repositories in our code You can use the helper method from the abstract class `getObjectModel` to help you build your `CustomerGroupRepository::get` method